Tests/Security/Http/Firewall/TwoFactorListenerTest.php

```diff
@@ -36,6 +36,7 @@ class TwoFactorListenerTest extends TestCase
     private const CHECK_PATH = '/check_path';
     private const AUTH_CODE_PARAM = 'auth_code_param';
     private const TRUSTED_PARAM = 'trusted_param';
+    private const PREFERED_PROVIDER_PARAM = 'preferProvider';
```

Comment:
If you make this a configurable parameter, you should use it in TwoFactorListener. But as I've mentioned about, I'd rather keep the logic in the controller, so that it can be overridden if needed.

Reply:
Outdated.